Pixel Game/Assets/Scripts/AiMovement.cs

```diff
@@ -0,0 +1,252 @@
+﻿using Pathfinding;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+public class AiMovement : Movement
+{
+    public enum TargetingStateEnum
+    {
+        Idle,
+        Retreat,
+        Follow,
+    }
+    [SerializeField] private AudioClip idleSound;
+    [SerializeField] private AudioClip retreatSound;
+    [SerializeField] private AudioClip followSound;
+
+    [SerializeField] private bool RoamWhenIdle;
+    [SerializeField] private float followUpperDistance = 4; //Stop following above this distance
+    [SerializeField] protected float followLowerDistance = 0; //Stop following belovw this distance
+    [SerializeField] private bool alwaysLookAtTarget;
+    [SerializeField] public Transform Target;
+    [SerializeField] protected float retreatUpperDistance = 0; //Stop retreating above this distance
+    private float roamStartTime;
+    private float roamDuration;
+    private const float MinRoamDuration = 2;
+    private const float MaxRoamDuration = 5;
+    private AiPath aiPath;
+    private TargetingStateEnum targetingState;
+
+    public TargetingStateEnum TargetingState 
+    { 
+        get => targetingState; 
+        private set
+        {
+            if (targetingState == value)
```

Comment:
What is value in this case?

Reply:
It's a reserved keyword in C# holding the value assigned to the property. So if you do `myObject.MyProperty = 5;` then `value` equals 5.

Reply:
Interesting !